### ISSUE TITLE
special character (Umlaut ö) not correctly shown on Android-Device (Odyssey) but correct on Puddletag

### ISSUE THREAD

**Issue by therob84** [puddletag/puddletag#593]
I am using a Umlaut (ö) within the field Artist, Title and also the filename does contain it - in Puddletag all is displayed correctly as "Jörg".
If I play the file in Android player Odyssey 1.1.21 (F-Droid) the filename is displayd correctly but if the mp3 tags are read 

- title contains: J**Ăś**rg (instead of J**ö**rg) 
- artist is correctly: Jörg 
- filename correctly: Jörg

In Strawberry player (Linux Mint) all is perfectly displayed as ö.

Could this be any error in my conversion procedure in puddletag or just any faulty setting? 
I used [this method](https://github.com/puddletag/puddletag/issues/592) for title conversion. 
Or could it be just a Odyseey bug? (But I never encountered this there until now - and also other files (not edited with puddletag) with äöü are not affected and displayed correctly).

Any hints? Thanks!

**Comment by therob84** [comment]
No hints or similar observations?

**Comment by blueblots** [comment]
Hi @therob84 , I tried to replicate this with Linux players but didn't see the problem. Could you let me know what format (mp3, aac etc.) your song file is?

It could have something to do with how Puddletag is writing a particular tag combining with whatever Odyssey is doing (my own baseless conjecture :-).

When I have time I'll try getting Odyssey and see if I can replicate this.

Thanks for reporting the issue in any case.

**Comment by therob84** [comment]
Hi,

meanwhile I just did some more testing today:

- I created a stupid MP3-file (440Hz noise) with Audacity and exported as MP3, included also some tag information (all filled with "Jörg") (..._original.mp3)
- I made a copy and edited it with puddletag (wrote some more "Jörgs" in some fields and saved it to the file ("Puddletag_Umlaut_Jörg.mp3"). 
- I then tested both mp3-files with various (mostly not recent) players I have installed on my android device (Lineage OS 15.1/FP2). 
- Result: For 3 different player: 2nd file edited with Puddletag is NOT recognized correctly, original file looks OK but the tags are not shown as intended. Only VLC makes it perfect. Any idea why?
----
- **Timber** (v.1.6): 
![grafik](https://user-images.githubusercontent.com/15350793/119565559-65816180-bdaa-11eb-9c75-a5cc5bf315b3.png)
- **Stock-LOS-Music-App** (for LOS 15.1 "com.cyanogenmod.eleven" v3.0): 
![grafik](https://user-images.githubusercontent.com/15350793/119565637-7cc04f00-bdaa-11eb-969e-8195941d00f7.png)
- **Odyssey** (v 1.1.21): 
![grafik](https://user-images.githubusercontent.com/15350793/119565755-a2e5ef00-bdaa-11eb-8c96-3ca387c223d4.png)
- Only **VLC** (v. 3.2.6) works : both files are displayed correctly!
![grafik](https://user-images.githubusercontent.com/15350793/119566074-007a3b80-bdab-11eb-9832-283f288430a3.png)
- Linux Mint 20.1 **Strawberry** (v 0.9.3) works also correct.

If I should test something else, let me know. Here are both files in a ZIP file: [PuddleTag_Umlaut.zip](https://github.com/puddletag/puddletag/files/6542244/PuddleTag_Umlaut.zip)

you can get Odyssey e.g. here: [F-Droid Repo](https://f-droid.org/en/packages/org.gateshipone.odyssey) / [latest APK](https://f-droid.org/repo/org.gateshipone.odyssey_33.apk).

In my original post the behaviour on different device is slightly different: no chinese character are shown but only the Umlaut is replaced with the stated faulty signs. Did not made tests to figure out this difference.

Let me know, if you have a glue what's going on here and what the fault could be.
best, Robert

**Comment by blueblots** [comment]
This may be related to #389

I'm just looking for any clues in older issues; I'll try fixing this at the end of the week if I've got time.

**Comment by blueblots** [comment]
Hi @therob84 , tonight I've been looking into this issue, and I've replicated it but right now I don't know why its happening or if its an issue with Puddletag.

I created my own test files (`jorg-test_original.mp3` and `jorg-test_copy.mp3`) using Audacity, then edited the `jorg-test_copy.mp3` file with Puddletag (added extra Jörgs to the fields). I tested them in VLC, Odyssey, and Foobar2000 (all Android versions), and the file which had been edited with Puddletag was showing up properly in all three programs. However when I tried viewing the files you uploaded in Odyssey, I saw the Chinese characters and the ? placeholder symbol, but VLC and Foobar2000 were able to display the titles properly... additionally all the Linux players I've tried can also display the umlaut properly. Some screenshots below:

My test file, edited with Puddletag and opened in Odyssey:
![screenshot1.jpg](https://user-images.githubusercontent.com/63152708/120550121-5c1c7880-c3ec-11eb-9447-a1d5c914c5c5.jpg)

Your uploaded file, opened in Odyssey:
![screenshot2.jpg](https://user-images.githubusercontent.com/63152708/120550165-66d70d80-c3ec-11eb-9a18-312fe5695b69.jpg)

This is probably a character encoding issue, most likely its a problem with Odyssey since other players don't seem to have an issue. But I don't know why my test file didn't encounter the same problem as yours.. could this be because of some system configuration/locale on your end?

Sorry I can't provide a fix right now, I'll have to do some more digging to get to the bottom of this. I'll let you know when I find a solution.
